backend/database/currencyDatabase.js

```diff
@@ -34,26 +35,39 @@ function getCurrencies() {
 
 // Adjust Currency
 // This adjust currency for a user. Can be given negative values. Provide it with the database record for a user.
-function adjustCurrency(user, currencyId, value) {
+function adjustCurrency(user, currencyId, value, adjustType = "adjust") {
     return new Promise(resolve => {
         if (!isViewerDBOn()) {
             return resolve();
         }
 
         // Dont do anything if value is not a number or is 0.
         value = parseInt(value);
-        if (isNaN(value) || value === 0) {
-            return resolve();
-        }
+        adjustType = adjustType.toLowerCase();
+        let newUserValue = value;
 
-        // Okay, move on and finish everything.
-        logger.debug(
-            "Currency: Adjusting " + value + " currency to " + user.username + ". " + currencyId
-        );
+        switch (adjustType) {
+        case "set":
```

Comment:
adjustType is new and allows us to pass in things like "set", "addPercentage", etc.. so we can calculate the new value differently.